pkg/gcp/actuator/actuator.go

```diff
@@ -426,9 +420,16 @@ func (a *Actuator) syncMint(ctx context.Context, cr *minterv1.CredentialsRequest
 			}
 			roles = append(roles, role.Name)
 		} else {
-			if !AreSlicesEqualWithoutOrder(role.IncludedPermissions, gcpSpec.Permissions) {
+			addedPermissions, removedPermissions := CalculateSliceDiff(role.IncludedPermissions, gcpSpec.Permissions)
+
+			if len(removedPermissions) > 0 {
```

Comment:
Just a note, I believe this will only be called if also there are added permissions, because otherwise needsUpdate will return false due to only checking for added permissions there.